source/gloperate-glfw/source/Window.cpp

```diff
@@ -472,25 +471,25 @@ void Window::handleEvent(WindowEvent & event)
 
 void Window::clearEventQueue()
 {
-    while (!m_eventQueue.empty())
-    {
-        delete m_eventQueue.front();
-        m_eventQueue.pop();
-    }
+    std::queue<std::unique_ptr<WindowEvent>> empty;
+    std::swap(m_eventQueue, empty);
 }
 
 bool Window::createInternalWindow(const GLContextFormat & format, int width, int height, GLFWmonitor * /*monitor*/)
 {
     // Abort if window is already created
-    assert(nullptr == m_context);
+    assert(!m_context);
     if (m_context)
     {
         return false;
     }
 
     // Create GLFW window with OpenGL context
     GLContextFactory factory;
-    m_context = static_cast<GLContext*>(factory.createBestContext(format));
+    m_context = std::unique_ptr<GLContext>(
+                static_cast<GLContext*>(factory.createBestContext(format).release())
```

Comment:
Actually this is a downcast from `AbstractGLContext` to `GLContext`. I found a solution [here](http://stackoverflow.com/questions/21174593/downcasting-unique-ptrbase-to-unique-ptrderived) and adapted it ignoring the delete (we use default deleter). Also we can use static_cast, because we know the gloperate_glfw::GLContextFactory creates a `GLContext`. If you find a cleaner way to do this, please let me know.